### PR TITLE
Implement computer player support in game flow

### PR DESCRIPTION
**Pull request by mackenziequinnjetton**
Please reference [Trello card](https://trello.com/c/Q3IjFovR/19-tic-tac-toe-implement-compplayer-into-game-loop-2pt)

Implement computer opponent support in main game class

## Added

- ComputerOpponent.incomplete_message

## Changed

- Player to HumanOpponent
- CompPlayer to ComputerOpponent
- Opponent class references, test file names, and test names to reflect the above two changes
- Game.play to call ComputerOpponent.incomplete_message if Game.opponent is a ComputerOpponent, then return :incomplete to end the game
- Require_relative statements in HumanOpponentTest and ComputerOpponentTests to require statements

## Misc

- Merge [this branch](https://github.com/mackenziequinnjetton/tic_tac_toe/pull/5) first

### CODE REVIEW

File: lib/computer_opponent.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class ComputerOpponent
+  def initialize; end
+
+  def get_move(*, board:)
+    rows = %i[row1 row2 row3]
+    cols = %i[col_a col_b col_c]
+    row_methods = [board.method(:row1), board.method(:row2), board.method(:row3)]
```
Comment (kevbuchanan) [inline lib/computer_opponent.rb:7]: I'm finding this hard to follow with the `method` captures. It's also kind of strange that this method needs to map `BoardRow` objects back to symbols like `:row1` and `:col_a`. Maybe those identifiers should be present on the `BoardRow` to make that easier? Then I believe the looping logic can just call the row and column methods directly:

```ruby
[board.row1, board.row2, board.row3].each do |row, row_index|
  [row.col_a, row.col_b, row.col_c].each do |space, col_index|
    return { row: row_index, col: col_index } if space == '-'
  end
end
```

Comment (mackenziequinnjetton) [reply lib/computer_opponent.rb]: This is outdated, as this functionality was included in this branch by accident and has now been removed. It is covered in [this PR](https://github.com/mackenziequinnjetton/tic_tac_toe/pull/7) instead. Please feel free to post the same feedback there and we can discuss it.

Comment (mackenziequinnjetton) [reply lib/computer_opponent.rb]: If you have no other proposed changes on this PR, would you please approve it?

**Review by kevbuchanan — APPROVED** [review]
